File: src/main/java/com/bajdcc/util/lexer/automata/dfa/DFAEdge.java
package com.bajdcc.util.lexer.automata.dfa;

/**
 * DFA边
 *
 * @author bajdcc
 */
public class DFAEdge {
	/**
	 * 初态
	 */
	public DFAStatus begin;

	/**
	 * 终态
	 */
	public DFAStatus end;

	/**
	 * 数据
	 */
	public DFAEdgeData data = new DFAEdgeData();
}

File: src/main/kotlin/com/bajdcc/util/lexer/automata/dfa/DFAEdge.kt
package com.bajdcc.util.lexer.automata.dfa

/**
 * DFA边
 *
 * @author bajdcc
 */
class DFAEdge {
    /**
     * 初态
     */
    var begin: DFAStatus? = null

    /**
     * 终态
     */
    var end: DFAStatus? = null

    /**
     * 数据
     */
    var data = DFAEdgeData()
}